ams/optimizer/src/main/java/com/netease/arctic/optimizer/FlinkOptimizerContainer.java

```diff
@@ -72,15 +82,105 @@ protected Map<String, String> doScaleOut(String startUpArgs) {
 
   @Override
   protected String buildOptimizerStartupArgsString(Resource resource) {
-    String taskManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(),
-        TASK_MANAGER_MEMORY_PROPERTY);
-    String jobManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(), JOB_MANAGER_MEMORY_PROPERTY);
+    Map<String, String> properties = resource.getProperties();
+    Map<String, String> flinkConfig = loadFlinkConfig();
+    Preconditions.checkState(properties != null && !flinkConfig.isEmpty(),
+        "resource properties is null or load flink-conf.yaml failed");
+    Long jobManagerMemory = getMemorySizeValue(properties, flinkConfig, JOB_MANAGER_MEMORY_PROPERTY,
+        JOB_MANAGER_TOTAL_PROCESS_MEMORY);
+    Long taskManagerMemory = getMemorySizeValue(properties, flinkConfig, TASK_MANAGER_MEMORY_PROPERTY,
+        TASK_MANAGER_TOTAL_PROCESS_MEMORY);
     String jobPath = getAMSHome() + "/plugin/optimize/OptimizeJob.jar";
-    long memory = Long.parseLong(jobManagerMemory) + Long.parseLong(taskManagerMemory) * resource.getThreadCount();
-    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s -c %s %s -m %s %s",
-        getFlinkHome(), taskManagerMemory, jobManagerMemory,
-        FlinkOptimizer.class.getName(), jobPath, memory,
-        super.buildOptimizerStartupArgsString(resource));
+    long memory = jobManagerMemory + taskManagerMemory * resource.getThreadCount();
+    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s %s -c %s %s -m %s %s",
+          getFlinkHome(), taskManagerMemory, jobManagerMemory, buildFlinkArgs(properties),
+          FlinkOptimizer.class.getName(), jobPath, memory,
+          super.buildOptimizerStartupArgsString(resource));
+  }
+
+  private Map<String, String> loadFlinkConfig() {
+    try {
+      return new Yaml().load(Files.newInputStream(Paths.get(getFlinkHome() + FLINK_CONFIG_PATH)));
```

Comment:
The way we can load configuration using flink `org.apache.flink.configuration.GlobalConfiguration#loadConfiguration(java.lang.String)`.

Reply:
> The way we can load configuration using flink `org.apache.flink.configuration.GlobalConfiguration#loadConfiguration(java.lang.String)`.
> 
> such as :
> 
> ```
>     Configuration flinkConfig = GlobalConfiguration.loadConfiguration(getFlinkHome() + FLINK_CONFIG_PATH);;
>     MemorySize jobMemorySize = flinkConfig.get(JobManagerOptions.TOTAL_PROCESS_MEMORY);
>     MemorySize taskMemorySize = flinkConfig.get(TaskManagerOptions.TOTAL_PROCESS_MEMORY);
> ```

Thank you very much for your suggestion. I originally thought of using the `flink api` to complete these operations, but later I looked at `pom.xml` and found that all flink related dependencies are `<scope>provided</scope>`. Therefore, if I use the `flink api` to do so, it will result in missing package dependencies.

I have considered the requirements for this module flink jars come from the local environment of flink, so I don't want to add flink dependencies to `pom.xml` to cause the package to become larger or have dependency conflicts, so I chose to implement the relevant functions myself.